// This file contains definitions of all configuration protos.
// These messages should only be stored in textual or in-memory form (i.e. not
// the binary format), so feed free to change any field numbers.

syntax = "proto2";

package heyp.proto;

option go_package = "github.com/uluyol/heyp-agents/go/proto";

enum Distribution {
  DIST_UNSET = 0;
  DIST_CONSTANT = 1;
  DIST_UNIFORM = 2;
  DIST_EXPONENTIAL = 3;
}

message TestLopriClientConfig {
  message WorkloadStage {
    optional int32 rpc_size_bytes = 1 [default = 4096];
    optional double target_average_bps = 2 [default = 10000000];
    optional Distribution interarrival_dist = 3 [default = DIST_EXPONENTIAL];
    optional string run_dur = 4 [default = "30s"];
  }

  optional int32 num_conns = 1 [default = 20];
  repeated WorkloadStage workload_stages = 2;
}
